docs/DISTRIBUTION.md

```diff
@@ -128,12 +128,50 @@ Each run creates all hourly and daily archives for the last 14 days. To prevent
+Both the application configuration as well as the Temporary Exposure Keys are signed so that the mobile devices
+can verify if the files are originating from the right backend. Using the signature, attack vectors like a
+man in the middle attack, request forgery etc. can be mitigated successfully. The implementation is done in
+[`AppConfigurationSigningDecorator`](/services/distribution/src/main/java/app/coronawarn/server/services/distribution/assembly/appconfig/structure/archive/decorator/signing/AppConfigurationSigningDecorator.java).
+
+The signing process takes the `export.bin` archive and creates a signature of the data structure which is then written into the `export.sig`.
+Please see`getSignatureFile` method located in [`SigningDecoratorOnDisk`](/services/distribution/src/main/java/app/coronawarn/server/services/distribution/assembly/structure/archive/decorator/signing/SigningDecoratorOnDisk.java).
+There are different implementations for that, i.e. how to sign an app configuration which is the
+`AppConfigurationSigningDecorator` - the specific code to determine how an application configuration file is supposed to be signed.
+For the Temporary Exposure Key files, they are signed in the similar way using the
+[`DiagnosisKeySigningDecorator`](/services/distribution/src/main/java/app/coronawarn/server/services/distribution/assembly/diagnosiskeys/structure/archive/decorator/signing/DiagnosisKeySigningDecorator.java).
+
+The algorithm used for signing the archives as well as other relevant information is defined in the [`application.yaml`](/services/distribution/src/main/resources/application.yaml).
 
 ## Bundling and Shifting
 
-Coming soon
+The [`DiagnosisKeyBundler`](/services/distribution/src/main/java/app/coronawarn/server/services/distribution/assembly/diagnosiskeys/DiagnosisKeyBundler.java)
+is responsible for the core logic of the distribution service, e.g. how the diagnosis keys are grouped into which date and which hour folder
+and necessary transformations.
+
+In the preparation step, the distribution job queries all diagnosis keys that are available in the database and
+cache them into the `DiagnosisKeyBundler`. The purpose of this class is to distribute keys into different hours and dates and tell
+the rest of the application for which hours and dates there are keys available.
+There are two of the key bundler, loaded based on the active profile.
+It is noteworthy to mention that the productive diagnosis key bundler applies a series of policies on the keys originating from the country specified by the parameter`origin-country`, which is defined in the [`application.yaml`](/services/distribution/src/main/resources/application.yaml).  Furthermore, with the parameter `apply-policies-for-all-countries` it is possible to apply the policies to all keys. The first policy,
+which is described in the GAEN framework (more details [here](https://developer.apple.com/documentation/exposurenotification/setting_up_a_key_server)),
```

Comment:
Policies are as follows:

- As described in the GAEN framework (more details [here](https://developer.apple.com/documentation/exposurenotification/setting_up_a_key_server)), diagnosis keys must not be distributed before two hours after the end of the key's expiration window.
- The shifting policy. Which ensures that each distribution / export file published to the CDN, contains at least 140 Temporary Exposure Keys (configurable by the `shifting-policy-threshold` property). When there are less keys available to distribute in a specific distribution run, these keys are shifted to a succeeding export, until the threshold minimum is fulfilled.